### ISSUE TITLE
Opening an unshared block shows an internal error rather than a proper UI

### ISSUE THREAD

**Issue by sunaurus** [warpdotdev/Warp#2631]
### Discord username (optional)

sunaurus#3806

### Describe the bug

When I open a link to a shared block that has been unshared, I see an unformatted page that just says `sql: no rows in result set`

### To Reproduce

1. Share a block and save the URL
2. Unshare the block
3. Open the URL 

### Expected behaviour

I expect to see some nice message like "Block not found, it may have been removed by the author", that way, users know that they are not doing anything wrong and the system is working properly.

### Screenshots

![image](https://user-images.githubusercontent.com/5356547/218059769-08aa089b-be9a-4ab4-ba77-ee52d91a443c.png)

### Operating System

MacOS

### Does this block you from using Warp daily?

No

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Yeah we've had some internal discussion of improving this, changing to a feature request for improved 404 experience.
